### PR TITLE
fix (rest) : Rest API for component search by Lucene search

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #2195 

Sample URL and response as follows in the screenshot. 

URL : http://localhost:8080/resource/api/components?name=comp&page=0&page_entries=20&luceneSearch=true&sort=name,desc

image

> Have you implemented any additional tests?
No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java [hunk 1]
```diff
@@ -141,6 +143,7 @@ public ResponseEntity<CollectionModel<EntityModel<Component>>> getComponents(
             @RequestParam(value = "fields", required = false) List<String> fields,
             @Parameter(description = "Flag to get components with all details.")
             @RequestParam(value = "allDetails", required = false) boolean allDetails,
+            @RequestParam(value = "luceneSearch", required = false) boolean luceneSearch,
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:146]: Add @Parameter same like other params

**Comment by hoangnt2** [comment]
Another thing, this snippet is unnecessary in lucene search case.

image

It also causes the bug that don't have enough entries on a page when we search by component type.
For example: I want a page with only 10 items. But when I search by Component Type some pages don't have enough entries.

I think the filter should be done before we create pagination result.
I am appreciate if you can help me to fix it.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java (hunk 1 above)
Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:147]: done

Comment (hoangnt2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:147]: @keerthi-bl could you take a look on my [comment](https://github.com/eclipse-sw360/sw360/pull/2201#issuecomment-1816034322)?

Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:148]: @hoangnt2 , I have addressed that and changed the snippet accordingly, please have a look.

**Comment by hoangnt2** [comment]
Change look good to me

**Pull request merged**